src/main/java/hudson/model/ExternalRun.java

```diff
@@ -170,16 +182,29 @@ public void cleanUp(BuildListener listener) {
         }
     }
 
+    /**
+     * @param result    Result code of the external job
+     * @param duration  Duration (in milliseconds) of the external job run
+     * @param stream    Stream of external job log
+     * @throws IOException
+     */
     public void acceptRemoteSubmission(final int result, final long duration, final InputStream stream) throws IOException {
         execute(new RunExecution() {
             public Result run(BuildListener listener) throws Exception {
-                PrintStream logger = new PrintStream(listener.getLogger());
+                PrintStream logger = new PrintStream(listener.getLogger(), false, "UTF-8");
                 final int sChunk = 8192;
                 GZIPInputStream zipin = new GZIPInputStream(stream);
                 byte[] buffer = new byte[sChunk];
                 int length;
-                while ((length = zipin.read(buffer, 0, sChunk)) != -1)
-                    logger.write(buffer, 0, length);
+                try {
+                    while ((length = zipin.read(buffer, 0, sChunk)) != -1)
+                        logger.write(buffer, 0, length);
+                } catch (Exception ex) {
+                    logger.print("An error occurred while reading the buffer: " + ex.getMessage());
+                    throw ex;
+                } finally {
+                    logger.close(); 
```

Comment:
Pretty much all of these changes were to satisfy something with findbugs. Suppressing is fine with me.